components/accelerator/nvidia/error/xid/component.go

```diff
@@ -5,150 +5,242 @@ package xid
+	cancel       context.CancelFunc
+	poller       query.Poller
+	currState    components.State
+	extraEventCh chan *components.Event
+	store        db.Store
+	mu           sync.RWMutex
+}
+
+func New(ctx context.Context, cfg nvidia_common.Config, dbRW *sql.DB, dbRO *sql.DB) *XIDComponent {
 	cfg.Query.SetDefaultsIfNotSet()
 	setDefaultPoller(cfg)
 
 	cctx, ccancel := context.WithCancel(ctx)
 	getDefaultPoller().Start(cctx, cfg.Query, nvidia_component_error_xid_id.Name)
 
-	return &component{
-		rootCtx: ctx,
-		cancel:  ccancel,
-		poller:  getDefaultPoller(),
+	extraEventCh := make(chan *components.Event, 256)
+	localStore, err := db.NewStore(dbRW, dbRO, "components_accelerator_nvidia_error_xid_events", DefaultRetentionPeriod)
+	if err != nil {
+		log.Logger.Errorw("failed to create store", "error", err)
+		ccancel()
+		return nil
+	}
+	return &XIDComponent{
+		rootCtx:      cctx,
+		cancel:       ccancel,
+		poller:       getDefaultPoller(),
+		extraEventCh: extraEventCh,
+		store:        localStore,
 	}
 }
 
-var _ components.Component = (*component)(nil)
-
-type component struct {
-	rootCtx context.Context
-	cancel  context.CancelFunc
-	poller  query.Poller
-}
-
-func (c *component) Name() string { return nvidia_component_error_xid_id.Name }
-
-// Just checks if the xid poller is working.
-func (c *component) States(_ context.Context) ([]components.State, error) {
-	return []components.State{
-		{
-			Name:    StateNameErrorXid,
-			Healthy: true,
-			Reason:  "xid event polling is working",
-		},
-	}, nil
-}
+var _ components.Component = (*XIDComponent)(nil)
 
-// tailScan fetches the latest output from the dmesg and the NVML poller
-// it is ok to call this function multiple times for the following reasons (thus shared with events method)
-// 1) dmesg "TailScan" is cheap (just tails the last x number of lines)
-// 2) NVML poller "Last" costs nothing, since we simply read the last state in the queue (no NVML call involved)
-func (c *component) tailScan() (*Output, error) {
-	dmesgC, err := components.GetComponent(dmesg.Name)
-	if err != nil {
-		return nil, err
-	}
+func (c *XIDComponent) Name() string { return nvidia_component_error_xid_id.Name }
 
-	var dmesgComponent *dmesg.Component
-	if o, ok := dmesgC.(interface{ Unwrap() interface{} }); ok {
-		if unwrapped, ok := o.Unwrap().(*dmesg.Component); ok {
-			dmesgComponent = unwrapped
-		} else {
-			return nil, fmt.Errorf("expected *dmesg.Component, got %T", dmesgC)
+func (c *XIDComponent) Start() error {
+	initializeBackoff := 1 * time.Second
+	for {
+		if err := c.updateCurrentState(); err != nil {
+			if strings.Contains(err.Error(), context.Canceled.Error()) {
+				log.Logger.Infow("context canceled, exiting")
+				return nil
+			}
+			log.Logger.Errorw("failed to fetch current events", "error", err)
+			time.Sleep(initializeBackoff)
+			continue
 		}
+		break
 	}
-	dmesgTailResults, err := dmesgComponent.TailScan()
+	watcher, err := pkg_dmesg.NewWatcher()
```

Comment:
Do we need `components.Start` explicitly when we can just call this in the https://github.com/leptonai/gpud/pull/321/files#diff-0de25174214578a8017687aeff4c98186ec09357955a44eec993c3a024f5ec5cR44?

```go
New(...
  go func() { if err := c.updateCurrentState(); err != nil {
  watcher, err := pkg_dmesg.NewWatcher()
  go watcher.start

Reply:
yes we do..created Start interface to guarantee ordering since now xid depends on os finish initializing